### PR TITLE
MB-414: Added endpoints createObserver with Id and getObserver with I…

### PR DESCRIPTION
**Pull request by ahdavies**
…d. Also removed subject references

### CODE REVIEW

File: mojio-sdk-rest/src/main/java/io/moj/java/sdk/MojioPushApi.java [hunk 1]
```diff
@@ -42,6 +42,32 @@
...
+     * {@link io.moj.java.sdk.Resource#MOJIO}, {@link io.moj.java.sdk.Resource#USER}, and
+     * {@link io.moj.java.sdk.Resource#VEHICLE} resources can be observed.
+     * @param resource String path for the specified resource. See {@link io.moj.java.sdk.Resource#getPath()}.
+     * @param resourceId the ID of the specified resource
+     * @return
+     * @see io.moj.java.sdk.Resource#getPath()
+     */
+    @GET("{resource}/{id}")
+    Call<Observer> getObserver(@Path("resource") String resource, @Path("id") String resourceId,
+                                  @Body Observer observer);
```
Comment (skidson) [inline mojio-sdk-rest/src/main/java/io/moj/java/sdk/MojioPushApi.java]: The GET call shouldn't have a Body :P

File: mojio-sdk-rest/src/main/java/io/moj/java/sdk/MojioPushApi.java [hunk 2]
```diff
@@ -42,6 +42,32 @@
...
+     * Creates an observer for the specified resource. Note: only
+     * {@link io.moj.java.sdk.Resource#MOJIO}, {@link io.moj.java.sdk.Resource#USER}, and
+     * {@link io.moj.java.sdk.Resource#VEHICLE} resources can be observed.
+     * @param resource String path for the specified resource. See {@link io.moj.java.sdk.Resource#getPath()}.
+     * @param resourceId the ID of the specified resource
+     * @return
+     * @see io.moj.java.sdk.Resource#getPath()
+     */
+    @POST("{resource}/{id}")
+    Call<Observer> createObserver(@Path("resource") String resource, @Path("id") String resourceId,
```
Comment (skidson) [inline mojio-sdk-rest/src/main/java/io/moj/java/sdk/MojioPushApi.java]: I think we'll also want the PUT("{resource}/{id}/{key}") at this time too - the main difference between PUT and POST is that PUT won't fail if the observer already exists and instead updates the existing one (useful for the app between upgrades, uninstalls, etc).

File: mojio-sdk-rest/src/main/java/io/moj/java/sdk/MojioPushApi.java [hunk 3]
```diff
@@ -51,8 +51,7 @@
      * @see io.moj.java.sdk.Resource#getPath()
      */
     @GET("{resource}/{id}")
-    Call<Observer> getObserver(@Path("resource") String resource, @Path("id") String resourceId,
-                                  @Body Observer observer);
+    Call<Observer[]> getObservers(@Path("resource") String resource, @Path("id") String resourceId);
```
Comment (skidson) [inline mojio-sdk-rest/src/main/java/io/moj/java/sdk/MojioPushApi.java]: Was this OK as an array? I think a List<Observer> like you had it is better if it works

Comment (ahdavies) [inline mojio-sdk-rest/src/main/java/io/moj/java/sdk/MojioPushApi.java]: It only works as an Array. Doesn't like being a List

**Comment by skidson** [comment]
Looks good!

**Pull request merged**